### DISCUSSION TITLE
Caddy seems to be open for arbitrary hostnames DNS challenges

### DISCUSSION THREAD

**Discussion by Surfict** [nextcloud/all-in-one#4882] [Questions] [answered]
### Steps to reproduce
1. Install Nextcloud AIO from scratch
2. After a few days, check the logs of nextcloud/all-in-one container

### Expected behavior 
Not having logs of ACME challenges coming from anyone on the internet

### Actual behavior 
I have logs of ACME challenges coming from external requests

### Host OS 
Ubuntu 22.04.4 

#### Nextcloud AIO version 
Nextcloud AIO v8.1.0
#### Other valuable info 
This issue is following [this discussion](https://github.com/nextcloud/all-in-one/discussions/4820) and [this post](https://caddy.community/t/unwanted-lets-encrypt-challenges-in-logs/24627/2) on Caddy forums. It seems that the way Caddy is currently configured "anyone across the internet can craft an SNI request for arbitrary hostnames on your server and prompt an ACME challenge from your Caddy instance. This leaves you open to abuse and should be rectified. Upstream ACME providers will have rate limits to mitigate your server abusing theirs, but you may find yourself with cluttered logs and have your renewal attempts rejected later due to said rate limit abuse."

This is indeed what I experienced myself.

**Comment by szaimen** [comment, marked as answer]
> It seems that the way Caddy is currently configured "anyone across the internet can craft an SNI request for arbitrary hostnames on your server and prompt an ACME challenge from your Caddy instance. Upstream ACME providers will have rate limits to mitigate your server abusing theirs, but you may find yourself with cluttered logs and have your renewal attempts rejected later due to said rate limit abuse."

Yes, this is by design.

As I answered already in https://github.com/nextcloud/all-in-one/discussions/4820, if you do not want such things to be logged anymore, I would recommend putting the aio interface behind a vpn and/or not exposing it publicly.

**Reply by Surfict** [reply]
Don't you think it should be at least mentioned somewhere, as by default this could lead people to have some problems renewing their domains ?

**Reply by szaimen** [reply]
Feel free to add a Point below FAQ in the readme that explains this behaviour.

**Reply by Surfict** [reply]
Done here : https://github.com/nextcloud/all-in-one/pull/4883

**Reply by devnoname120** [reply]
> Yes, this is by design.

@szaimen Could you expand on the reasoning behind this? Thanks

**Comment by rugk** [comment]
Hmm, is not this effectively a (D)DOS vulnerability kinda? I mean, it's not just "something annoying is logged". If I spam your server with invalid domain requests for three months or so, Caddy will at some point not be able to renew your _actually used cert(s)_, it will expire and your server won't be reachable anymore?

I mean, as a proposal: Maybe an env variable or so to configure this and only accept one domain name would be useful? I guess this also covers 90% of the use cases, IMHO it's unlikely people want to have their Nextcloud available under different domains.
